src/main/webapp/feed.html

```diff
@@ -23,24 +23,48 @@ <h1>Message Feed</h1>
 
     function MessageFeed() {
       const [allMessages, setMessages] = React.useState([])
+      const [numMessagesPerUser, setNumUserMessages] = React.useState(["No messages!", h('br')])
 
       React.useEffect(fetchMessages, []);
 
       // Fetch messages and add them to the page.
       function fetchMessages() {
+        // store message react components
         const fetchedMessages = []
+        // store how many messages each user posts in map for later use
+        const userMessageCounts = {}
+        // store react components of the num of msgs each user posts
+        const finalUserMessageCounts = ["Number of Messages Per User:", h('br')]
         const url = '/feed';
         fetch(url)
           .then(response => response.json())
           .then(messages => {
+            // get data from each message
             messages.forEach(message => {
+              // if a user has posted before, incr
+              if (message.user in userMessageCounts) userMessageCounts[message.user]++
+              // otherwise set their msg counts to 1
+              else userMessageCounts[message.user] = 1
+              // push message react component to list
               fetchedMessages.push(buildMessageDiv(message));
             });
+          })
+          .then(() => {
+            // update render state of all messages posted
             setMessages(fetchedMessages);
+            // go thru each user and their counts and push a react component to the list
+            for(user in userMessageCounts)
+              finalUserMessageCounts.push(buildUserCountDiv(user, userMessageCounts[user]))
+          })
+          .then(() => {
+            // update render state of each users message counts if msgs pushed
+            if (finalUserMessageCounts > 2)
```

Comment:
This check is fragile. What would happen if another person edited your code and removed an element from the initial value of this array? Then you wouldn't display any message counts.

One thing you can do is extract the initial array's value into a constant and call `.length` here. But there is also probably a better way to structure your code.